src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py

```diff
@@ -491,40 +507,84 @@ def export_coreml(self, filename):
-        input_name = feature_extractor.data_layer
         output_name = 'distance'
-        input_features = [(input_name, _datatypes.Array(*(self.input_image_shape)))]
         output_features = [(output_name, _datatypes.Array(num_examples))]
 
-        # Create a neural network
-        builder = _neural_network.NeuralNetworkBuilder(
-            input_features, output_features, mode=None)
-
-        # Convert the feature extraction network
-        mx_feature_extractor = feature_extractor._get_mx_module(
-            feature_extractor.ptModel.mxmodel,
-            feature_extractor.data_layer,
-            feature_extractor.feature_layer,
-            feature_extractor.context,
-            self.input_image_shape
-        )
-        batch_input_shape = (1, ) + self.input_image_shape
-        _mxnet_converter.convert(mx_feature_extractor, mode=None,
-                                 input_shape=[(input_name, batch_input_shape)],
-                                 builder=builder, verbose=False)
+        if self.model != 'VisionFeaturePrint_Screen':
+            # Convert the MxNet model to Core ML
+            ptModel = _pre_trained_models.MODELS[self.model]()
+            feature_extractor = _image_feature_extractor.MXFeatureExtractor(ptModel)
+
+            input_name = feature_extractor.data_layer
+            input_features = [(input_name, _datatypes.Array(*(self.input_image_shape)))]
+
+            # Create a neural network
+            builder = _neural_network.NeuralNetworkBuilder(
+                input_features, output_features, mode=None)
+
+            # Convert the feature extraction network
+            mx_feature_extractor = feature_extractor._get_mx_module(
+                feature_extractor.ptModel.mxmodel,
+                feature_extractor.data_layer,
+                feature_extractor.feature_layer,
+                feature_extractor.context,
+                self.input_image_shape
+            )
+            batch_input_shape = (1, ) + self.input_image_shape
+            _mxnet_converter.convert(mx_feature_extractor, mode=None,
+                                     input_shape=[(input_name, batch_input_shape)],
+                                     builder=builder, verbose=False)
+            feature_layer = feature_extractor.feature_layer
+
+        else:     # self.model == VisionFeaturePrint_Screen
+            # Create a pipleline that contains a VisionFeaturePrint followed by a
+            # neural network.
+            BGR_VALUE = _cmt.proto.FeatureTypes_pb2.ImageFeatureType.ColorSpace.Value('BGR')
+            DOUBLE_ARRAY_VALUE = _cmt.proto.FeatureTypes_pb2.ArrayFeatureType.ArrayDataType.Value('DOUBLE')
+
+            top_spec = _cmt.proto.Model_pb2.Model()
+            top_spec.specificationVersion = 3
+            desc = top_spec.description
+
+            input = desc.input.add()
+            input.name = self.feature
+            input.type.imageType.width = 299
+            input.type.imageType.height = 299
```

Comment:
Do we need to repeat 299 here? We already stored this inside `input_image_shape`.

Reply:
Ok. Done.